engine/table/src/main/java/io/deephaven/engine/table/impl/select/SelectColumnWithRespectsBarrier.java

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.impl.select;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.function.Predicate;
+
+/**
+ * A SelectColumn that respects the given barriers.
+ */
+public class SelectColumnWithRespectsBarrier extends WrappedSelectColumn {
+    private final Object[] respectedBarriers;
+
+    /**
+     * Return a new SelectColumn that respects the given barriers.
+     * 
+     * @param toWrap SelectColumn to wrap
+     * @param respectBarriers the barriers to respect
+     * @return a new select colum that respects the given barriers.
+     */
+    public static SelectColumn addRespectsBarriers(SelectColumn toWrap, Object... respectBarriers) {
+        return new SelectColumnWithRespectsBarrier(toWrap, true, respectBarriers);
```

Comment:
In this and similar methods, should we return `toWrap` is `respectsBarriers` is null or empty?